File: day2_python/part2.py
def parse_line(line):
  cmd, step = line.split(" ")
  return (cmd, int(step))

file = open("input.txt", "r")
input = file.read()
file.close()

depth = 0
x = 0
aim = 0
for (cmd, step) in map(parse_line, input.splitlines()):
  if cmd == "forward":
    x = x + step
    depth = depth + aim * step
  elif cmd == "down":
    aim = aim + step
  elif cmd == "up":
    aim = aim - step

result = x * depth
print(result)

File: day6_ruby/part2.rb
fishes = File.read("input.txt").split(",").map(&:to_i).inject(Array.new(9, 0)) {|l, e| l[e] += 1; l}

256.times { |i|
  fishes[(i + 7) % fishes.length] += fishes[i % fishes.length]
}

puts fishes.inject(0, :+)